analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -3,617 +3,391 @@
 using System.Collections.ObjectModel;
 using System.Collections.Specialized;
 using System.Linq;
+using System.Runtime.Serialization;
 using System.Text;
 
-class ListTests
+class CollectionTests
 {
     private IEnumerable<int> items = new List<int> { 1, 2, 3 };
+    private IDictionary<int, int> dictionaryItems = new Dictionary<int, int>();
 
     private static bool Predicate(int i) => true;
     private static void Action(int i) { }
-    private List<int> Factory() => new List<int>();
+    private List<int> ListFactory() => new List<int>();
+    private HashSet<int> SetFactory() => new HashSet<int>();
+    private Queue<int> QueueFactory() => new Queue<int>();
+    private Stack<int> StackFactory() => new Stack<int>();
+    private ObservableCollection<int> ObservableCollectionFactory() => new ObservableCollection<int>();
+    private int[] ArrayFactory() => new int[0];
+    private Dictionary<int, int> DictionaryFactory() => new Dictionary<int, int>();
```

Comment:
I would name these `CreateXxx`, or `GetXxxx` (and return null, as the value has no effect on the UT.
So
```
private List<int> CreateList() => new List<int>();
```
or
```
private List<int> CreateList() => null; // Syntax sugar, irrelevant for the test cases
```
or (even I have personal anti-preference for Get, in this case, it makes a little bit of sense)
```
private List<int> GetList() => null; 
```